### PR TITLE
fix: (Low-Code) (DeclarativeOAuthFlow) - fix `tooltip examples` and `labels` representation to show values instead of full object

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Based on this review comment, there is a need to fix how tooltip are represented in order not to confuse Customers.

https://github.com/airbytehq/airbyte-platform-internal/pull/14785#discussion_r1881195226

## How
- edited the `declarative_component_schema.yaml`
- rebuild the models

## User Impact
No impact is expected. Not a breaking change.

## Summary by CodeRabbit

- **New Features**
  - Introduced new components: `ComponentMappingDefinition`, `HttpComponentsResolver`, and `StreamConfig` for enhanced data handling and configuration.

- **Updates**
  - Enhanced `OAuthConfigSpecification` with clearer descriptions and simplified examples.
  - Refined `DeclarativeStream` and `DynamicDeclarativeStream` for better clarity on their functionalities.

- **Improvements**
  - Expanded interpolation section with new macros and filters for dynamic data management.
  - Streamlined example values in `OauthConnectorInputSpecification` for improved readability and consistency.

### CODE REVIEW

**Review by bnchrch — APPROVED** [review]

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 1]
```diff
@@ -2203,49 +2195,49 @@ definitions:
               max:
                 type: integer
             examples:
-              - state: { "min": 7, "max": 128 }
+              - { "min": 7, "max": 128 }
           client_id_key:
-            title: (Optional) DeclarativeOAuth Client ID Key Override
+            title: (Optional) Client ID Key Override
```
Comment (lmossman) [inline airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2200]:
```suggestion
            title: Client ID Key Override
```

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 2]
```diff
@@ -2203,49 +2195,49 @@ definitions:
...
             type: string
             description: |-
               The DeclarativeOAuth Specific optional override to provide the custom `client_id` key name, if required by data-provider.
             examples:
-              - client_id_key: "my_custom_client_id_key_name"
+              - "my_custom_client_id_key_name"
           client_secret_key:
-            title: (Optional) DeclarativeOAuth Client Secret Key Override
+            title: (Optional) Client Secret Key Override
```
Comment (lmossman) [inline airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2207]:
```suggestion
            title: Client Secret Key Override
```

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 3]
```diff
@@ -2203,49 +2195,49 @@ definitions:
...
             type: string
             description: |-
               The DeclarativeOAuth Specific optional override to provide the custom `client_secret` key name, if required by data-provider.
             examples:
-              - client_secret_key: "my_custom_client_secret_key_name"
+              - "my_custom_client_secret_key_name"
           scope_key:
-            title: (Optional) DeclarativeOAuth Scope Key Override
+            title: (Optional) Scope Key Override
```
Comment (lmossman) [inline airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2214]:
```suggestion
            title: Scope Key Override
```

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 4]
```diff
@@ -2203,49 +2195,49 @@ definitions:
...
             type: string
             description: |-
               The DeclarativeOAuth Specific optional override to provide the custom `scope` key name, if required by data-provider.
             examples:
-              - scope_key: "my_custom_scope_key_key_name"
+              - "my_custom_scope_key_key_name"
           state_key:
-            title: (Optional) DeclarativeOAuth State Key Override
+            title: (Optional) State Key Override
```
Comment (lmossman) [inline airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2221]:
```suggestion
            title: State Key Override
```

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 5]
```diff
@@ -2203,49 +2195,49 @@ definitions:
...
             type: string
             description: |-
               The DeclarativeOAuth Specific optional override to provide the custom `state` key name, if required by data-provider.
             examples:
-              - state_key: "my_custom_state_key_key_name"
+              - "my_custom_state_key_key_name"
           auth_code_key:
-            title: (Optional) DeclarativeOAuth Auth Code Key Override
+            title: (Optional) Auth Code Key Override
```
Comment (lmossman) [inline airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2228]:
```suggestion
            title: Auth Code Key Override
```

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 6]
```diff
@@ -2203,49 +2195,49 @@ definitions:
...
             type: string
             description: |-
               The DeclarativeOAuth Specific optional override to provide the custom `code` key name to something like `auth_code` or `custom_auth_code`, if required by data-provider.
             examples:
-              - auth_code_key: "my_custom_auth_code_key_name"
+              - "my_custom_auth_code_key_name"
           redirect_uri_key:
-            title: (Optional) DeclarativeOAuth Redirect URI Key Override
+            title: (Optional) Redirect URI Key Override
```
Comment (lmossman) [inline airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2235]:
```suggestion
            title: Redirect URI Key Override
```

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 7]
```diff
@@ -2130,65 +2130,57 @@ definitions:
...
               The DeclarativeOAuth Specific string URL string template to initiate the authentication.
               The placeholders are replaced during the processing to provide neccessary values.
             examples:
-              - consent_url: https://domain.host.com/marketing_api/auth?{client_id_key}={{client_id_key}}&{redirect_uri_key}={urlEncoder:{{redirect_uri_key}}}&{state_key}={{state_key}}
-              - consent_url: https://endpoint.host.com/oauth2/authorize?{client_id_key}={{client_id_key}}&{redirect_uri_key}={urlEncoder:{{redirect_uri_key}}}&{scope_key}={urlEncoder:{{scope_key}}}&{state_key}={{state_key}}&subdomain={subdomain}
+              - https://domain.host.com/marketing_api/auth?{client_id_key}={{client_id_key}}&{redirect_uri_key}={urlEncoder:{{redirect_uri_key}}}&{state_key}={{state_key}}
+              - https://endpoint.host.com/oauth2/authorize?{client_id_key}={{client_id_key}}&{redirect_uri_key}={urlEncoder:{{redirect_uri_key}}}&{scope_key}={urlEncoder:{{scope_key}}}&{state_key}={{state_key}}&subdomain={subdomain}
           scope:
-            title: (Optional) DeclarativeOAuth Scope
+            title: (Optional) Scope
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2142]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 8]
```diff
@@ -2130,65 +2130,57 @@ definitions:
...
             type: string
             description: |-
               The DeclarativeOAuth Specific URL templated string to obtain the `access_token`, `refresh_token` etc.
               The placeholders are replaced during the processing to provide neccessary values.
             examples:
-              - access_token_url: https://auth.host.com/oauth2/token?{client_id_key}={{client_id_key}}&{client_secret_key}={{client_secret_key}}&{auth_code_key}={{auth_code_key}}&{redirect_uri_key}={urlEncoder:{{redirect_uri_key}}}
+              - https://auth.host.com/oauth2/token?{client_id_key}={{client_id_key}}&{client_secret_key}={{client_secret_key}}&{auth_code_key}={{auth_code_key}}&{redirect_uri_key}={urlEncoder:{{redirect_uri_key}}}
           access_token_headers:
-            title: (Optional) DeclarativeOAuth Access Token Headers
+            title: (Optional) Access Token Headers
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2157]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 9]
```diff
@@ -2130,65 +2130,57 @@ definitions:
...
               The DeclarativeOAuth Specific optional query parameters to inject while exchanging the `auth_code` to `access_token` during `completeOAuthFlow` step.
               When this property is provided, the query params will be encoded as `Json` and included in the outgoing API request.
             examples:
-              - access_token_params:
-                  {
-                    "{auth_code_key}": "{{auth_code_key}}",
-                    "{client_id_key}": "{{client_id_key}}",
-                    "{client_secret_key}": "{{client_secret_key}}",
-                  }
+              - {"{client_id_key}": "{{client_id_key}}"}
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2172]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 1 above)
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2200]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 2 above)
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2207]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 3 above)
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2214]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 4 above)
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2221]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 5 above)
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2228]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml (hunk 6 above)
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml:2235]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 10]
```diff
@@ -2130,65 +2130,57 @@ definitions:
...
             items:
               type: string
             description: |-
               The DeclarativeOAuth Specific list of strings to indicate which keys should be extracted and returned back to the input config.
             examples:
-              - extract_output: ["access_token", "refresh_token", "other_field"]
+              - ["access_token", "refresh_token", "other_field"]
           state:
-            title: (Optional) DeclarativeOAuth Configurable State Query Param
+            title: (Optional) Configurable State Query Param
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml]: Done.

File: airbyte_cdk/sources/declarative/declarative_component_schema.yaml [hunk 11]
```diff
@@ -2130,65 +2130,57 @@ definitions:
...
               The DeclarativeOAuth Specific optional headers to inject while exchanging the `auth_code` to `access_token` during `completeOAuthFlow` step.
             examples:
-              - access_token_headers:
-                  {
-                    "Authorization": "Basic {base64Encoder:{client_id}:{client_secret}}",
-                  }
+              - {"Authorization": "Basic {base64Encoder:{client_id}:{client_secret}}"}
           access_token_params:
-            title: (Optional) DeclarativeOAuth Access Token Query Params (Json Encoded)
+            title: (Optional) Access Token Query Params (Json Encoded)
```
[the remark this answers is not in the document]
Comment (bazarnov) [reply airbyte_cdk/sources/declarative/declarative_component_schema.yaml]: Done.

**Review by lmossman — APPROVED** [review]

**Pull request merged**